### ISSUE TITLE
Legend not announced when following a link to main region using Ctrl-Option-Space

### ISSUE THREAD

**Issue by MartinJJones** [alphagov/reported-bugs#74]
**Upstream bug:** [https://bugs.webkit.org/show_bug.cgi?id=246607](https://bugs.webkit.org/show_bug.cgi?id=246607)
**Date:** 2022-10-17
**Reported by:** Martin Jones
**Related to:** Similar to the issue reported on https://bugs.webkit.org/show_bug.cgi?id=191742, main difference being the `aria-describedby` value is read out when the link points to an input element, but nothing is read out in the scenario below, even when `aria-describedby` is used.

---

### Overview

Load the HTML page attached and use the VoiceOver Ctrl-Option-Space shortcut on the "Skip to main content" link to jump to the main content section.

### Reduced test case

```HTML
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>

<body>
  <a href="#main-content">Skip to main content</a>
  <main id="main-content">
    <fieldset>
      <legend>
        <h1>Heading Question?</h1>
      </legend>

      <input type="radio" name="response" id="response-0" value="yes">
      <label for="response-0">Yes</label>

      <input type="radio" name="response" id="response-1" value="no">
      <label for="response-1">No</label>
    </fieldset>
  </main>
</body>

</html>
```

### Expected result

VoiceOver should read the heading text "VoiceOver will read out the legend h1 text "Heading Question?"

### Actual result

Nothing is read out

https://user-images.githubusercontent.com/28779939/196145190-efa8d4b8-81cd-4077-b316-ea183612aa73.mov
